public/API/API_GetGameList.php

```diff
@@ -31,22 +33,52 @@
 
 $withAchievements = (bool) request()->query('f');
 $withHashes = (bool) request()->query('h');
+$offset = (int) request()->query('o');
+$count = (int) request()->query('c');
 
-getGamesListByDev(null, $consoleID, $dataOut, 1, false, $withAchievements ? 0 : 2);
+$query = DB::table('GameData')
+    ->leftjoin('Console AS c', 'c.ID', '=', 'GameData.ConsoleID')
+    ->leftjoin('Achievements AS ach', 'ach.GameID', '=', 'GameData.ID')
+    ->leftjoin('LeaderboardDef AS ld', 'ld.GameID', '=', 'GameData.ID')
+    ->select(
+        'GameData.*',
+        'c.Name as ConsoleName',
+        DB::raw('MAX(ach.DateModified) as DateModified'),
+        DB::raw('COALESCE(GameData.achievements_published,0) AS NumAchievements'),
+        DB::raw('COUNT(DISTINCT ld.ID) AS NumLBs')
+    )
+    ->where('ConsoleID', $consoleID);
+
+if ($withAchievements) {
+    $query
+        ->where('achievements_published', '>', 0);
+}
+
+if ($offset > 0 && $count > 0) {
+    $query
+        ->offset($offset)
+        ->take($count);
+}
+
+$queryResponse = $query
+    ->groupBy('GameData.ID')
+    ->orderBy('GameData.Title', 'asc')
+    ->get();
```

Comment:
Verified this is now working as expected. I believe some of this conditional logic can be chained onto the original base query and still be functional:

```php
$query = DB::table('GameData')
    ->leftJoin('Console AS c', 'c.ID', '=', 'GameData.ConsoleID')
    ->leftJoin('Achievements AS ach', 'ach.GameID', '=', 'GameData.ID')
    ->leftJoin('LeaderboardDef AS ld', 'ld.GameID', '=', 'GameData.ID')
    ->select(
        'GameData.*',
        'c.Name as ConsoleName',
        DB::raw('MAX(ach.DateModified) as DateModified'),
        DB::raw('COALESCE(GameData.achievements_published, 0) AS NumAchievements'),
        DB::raw('COUNT(DISTINCT ld.ID) AS NumLBs')
    )
    ->where('GameData.ConsoleID', $consoleID)
    ->when($withAchievements, function ($query) {
        return $query->where('GameData.achievements_published', '>', 0);
    })
    ->groupBy('GameData.ID')
    ->orderBy('GameData.Title', 'asc');

if ($offset > 0 && $count > 0) {
    $query->offset($offset)->take($count);
}

$queryResponse = $query->get();
```